pkg/cmd/admin/router/router.go

```diff
@@ -259,28 +342,40 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			insecure = "true"
 		}
 
-		defaultCert, err := loadDefaultCert(cfg.DefaultCertificate)
+		defaultCert, err := loadCert(cfg.DefaultCertificate)
 		if err != nil {
 			return fmt.Errorf("router could not be created; error reading default certificate file", err)
 		}
 
+		privateKey, err := loadKey(cfg.ExternalHostPrivateKey)
+		if err != nil {
+			return fmt.Errorf("router could not be created; error reading private key for external host", err)
+		}
+
 		if len(cfg.StatsPassword) == 0 {
 			cfg.StatsPassword = generateStatsPassword()
 			fmt.Fprintf(out, "password for stats user %s has been set to %s\n", cfg.StatsUsername, cfg.StatsPassword)
 		}
 
 		env := app.Environment{
-			"OPENSHIFT_MASTER":         config.Host,
-			"OPENSHIFT_CA_DATA":        string(config.CAData),
-			"OPENSHIFT_KEY_DATA":       string(config.KeyData),
-			"OPENSHIFT_CERT_DATA":      string(config.CertData),
-			"OPENSHIFT_INSECURE":       insecure,
-			"DEFAULT_CERTIFICATE":      defaultCert,
-			"ROUTER_SERVICE_NAME":      name,
-			"ROUTER_SERVICE_NAMESPACE": namespace,
-			"STATS_PORT":               strconv.Itoa(cfg.StatsPort),
-			"STATS_USERNAME":           cfg.StatsUsername,
-			"STATS_PASSWORD":           cfg.StatsPassword,
+			"OPENSHIFT_MASTER":                   config.Host,
+			"OPENSHIFT_CA_DATA":                  string(config.CAData),
+			"OPENSHIFT_KEY_DATA":                 string(config.KeyData),
+			"OPENSHIFT_CERT_DATA":                string(config.CertData),
+			"OPENSHIFT_INSECURE":                 insecure,
+			"DEFAULT_CERTIFICATE":                defaultCert,
+			"ROUTER_SERVICE_NAME":                name,
+			"ROUTER_SERVICE_NAMESPACE":           namespace,
+			"ROUTER_EXTERNAL_HOST_HOSTNAME":      cfg.ExternalHost,
+			"ROUTER_EXTERNAL_HOST_USERNAME":      cfg.ExternalHostUsername,
+			"ROUTER_EXTERNAL_HOST_PASSWORD":      cfg.ExternalHostPassword,
+			"ROUTER_EXTERNAL_HOST_HTTP_VSERVER":  cfg.ExternalHostHttpVserver,
+			"ROUTER_EXTERNAL_HOST_HTTPS_VSERVER": cfg.ExternalHostHttpsVserver,
+			"ROUTER_EXTERNAL_HOST_INSECURE":      strconv.FormatBool(cfg.ExternalHostInsecure),
+			"ROUTER_EXTERNAL_HOST_PRIVKEY":       privateKey,
```

Comment:
I have fixed this in an upcoming fixups commit.  I also had to update the integration tests to do the same bind-mount thing.